File: platform-lib-go/dto.go
// Package platformlib is a minimal Go port of the PLM Core platform-lib.
// It reproduces the wire contracts (JWT claims, registration DTOs, instance-id
// derivation, NATS subjects, S2S headers) so Go services interoperate with the
// Java stack. Only the subset needed by thin services (ws-gateway) is covered:
// JWT verification, the local registry, self-registration and a NATS wrapper.
package platformlib

// RegisterRequest is POSTed to platform-api
// /api/platform/internal/environment/register. Mirrors the Java record;
// JSON field names match Jackson's camelCase output.
type RegisterRequest struct {
	ServiceCode string   `json:"serviceCode"`
	BaseURL     string   `json:"baseUrl"`
	HealthURL   string   `json:"healthUrl"`
	RoutePrefix string   `json:"routePrefix"`
	ExtraPaths  []string `json:"extraPaths"`
	Version     string   `json:"version"`
	SpaceTag    string   `json:"spaceTag"`
	Features    []string `json:"features"`
}

// ServiceInstanceInfo is the lightweight per-instance view in a snapshot.
type ServiceInstanceInfo struct {
	InstanceID  string `json:"instanceId"`
	ServiceCode string `json:"serviceCode"`
	BaseURL     string `json:"baseUrl"`
	Version     string `json:"version,omitempty"`
	SpaceTag    string `json:"spaceTag,omitempty"`
	Healthy     bool   `json:"healthy"`
}

// RegistrySnapshot is pulled from platform-api. Version is monotonic.
type RegistrySnapshot struct {
	Version  int64                            `json:"version"`
	Services map[string][]ServiceInstanceInfo `json:"services"`
}

// RegisterResponse is the body returned by a successful registration.
type RegisterResponse struct {
	InstanceID string `json:"instanceId"`
}

File: platform-lib-zig/src/dto.zig
const std = @import("std");

/// POSTed to platform-api /internal/environment/register. Field names match
/// Jackson camelCase so std.json.stringify produces the Java-compatible body.
pub const RegisterRequest = struct {
    serviceCode: []const u8,
    baseUrl: []const u8,
    healthUrl: []const u8,
    routePrefix: []const u8,
    extraPaths: []const []const u8,
    version: []const u8,
    spaceTag: []const u8,
    features: []const []const u8,
};

/// Body returned by a successful registration.
pub const RegisterResponse = struct {
    instanceId: []const u8 = "",
};

/// Lightweight per-instance view in a registry snapshot.
pub const ServiceInstanceInfo = struct {
    instanceId: []const u8,
    serviceCode: []const u8,
    baseUrl: []const u8,
    version: []const u8 = "",
    spaceTag: []const u8 = "",
    healthy: bool = false,
};

test "RegisterRequest stringifies to camelCase JSON" {
    const alloc = std.testing.allocator;
    const req = RegisterRequest{
        .serviceCode = "dst",
        .baseUrl = "http://dst:8086",
        .healthUrl = "http://dst:8086/actuator/health",
        .routePrefix = "/api/dst/**",
        .extraPaths = &.{},
        .version = "0.1.0",
        .spaceTag = "",
        .features = &.{},
    };
    const json = try std.fmt.allocPrint(alloc, "{f}", .{std.json.fmt(req, .{})});
    defer alloc.free(json);
    try std.testing.expect(std.mem.indexOf(u8, json, "\"serviceCode\":\"dst\"") != null);
    try std.testing.expect(std.mem.indexOf(u8, json, "\"routePrefix\":\"/api/dst/**\"") != null);
}
